### PR TITLE
Issues/207

### PR DESCRIPTION
**Pull request by jccaicedo**
Functional loading of pre-trained weights for ResNet and DenseNet models.
Loading weights is slow and can be optimized in the future, but it's currently working well.
Pre-trained weights add the benefit of improved accuracy due to a better initialization.

### CODE REVIEW

**Comment by jccaicedo** [comment]
The latest commit fixes tests and addresses one of the questions in issues #196

**Comment by Arkkienkeli** [comment]
Hi, I have started reviewing, so far I have these questions:
1. Why would you like to change `keras>=2.2.4` to `keras>=2.2.5` ? Does anything prevent us from using the older version here?
2. You have removed `test_define_model` from `test_densenet.py`, but you did not remove the same test from `test_resnet.py`.
3. In `resnet.py` thew code was reorganized, but in `densenet.py` most of the code was removed (but densenet works in a similar way as resnet).
4. In `model.py` what is the purpose of `validation_freq=freq`  (line 80)?

I am going to try it locally.
Thank you!

**Comment by jccaicedo** [comment]
Great questions! Here are my comments:

1. According to the release history of Keras, version `2.2.5` is the last that doesn't introduce breaking API changes, and also incorporated new ResNet architectures in the Applications module. I preferred to use the naming convention of `2.2.5` for ResNet because it is consistent with DenseNet. In `2.2.4` the naming was not uniform.
2. I tried to fix the tests for DenseNet and ResNet which are currently breaking. I am not quite sure why, contributions here are welcome.
3. That's correct. I reorganized the code of ResNet to make it reusable for DenseNet, specially the weights initialization part. Turns out, all the code is reusable if we use `keras==2.2.5`. In that way, the plugin for DenseNet inherits all the functionalities of the plugin for ResNet (it is a subclass) and only needs to define the Keras Application models that are supported. We can use the same pattern for other CNN architectures as long as they are supported in Keras Applications.
4. I introduced changes to `image_dataset.py` to remove the `load_factor` variable, which was redundant with `sampling_factor` in the config file. Given that we can control the sampling factor, making epochs shorter, we may want to reduce the validation frequency to avoid overhead computation time.

Thanks for the review! Let me know if there is something else that needs clarification or change.

**Comment by Arkkienkeli** [comment]
We can merge, I guess.
